client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/TransactionContext.java

```diff
@@ -69,17 +69,17 @@ public String toString() {
              '}';
    }
 
-   boolean containsKey(Object key, Function<K, MetadataValue<V>> remoteValueSupplier) {
-      ByRef<Boolean> result = new ByRef<>(null);
+   CompletableFuture<Boolean> containsKey(Object key, Function<K, MetadataValue<V>> remoteValueSupplier) {
```

Comment:
use `CompletionStage` instead? I think it is the recommended way to do it.

Reply:
Unfortunately, AsyncCache defines containsKeyAsync as returning a CompletableFuture :(